tst/EnergyPlus/unit/Humidifiers.unit.cc

```diff
@@ -4,411 +4,245 @@
+			"  Mixed Air Node 1,        !- Air Inlet Node Name",
+			"  Main Humidifier Outlet Node,  !- Air Outlet Node Name",
+			"  ;                        !- Water Storage Tank Name",
+			"  Curve:Cubic,",
+			"    ThermalEfficiencyFPLR,   !- Name",
+			"    1.0,                     !- Coefficient1 Constant",
+			"    0.0,                     !- Coefficient2 x",
+			"    0.0,                     !- Coefficient3 x**2",
+			"    0.0,                     !- Coefficient4 x**3",
+			"    0.0,                     !- Minimum Value of x",
+			"    1.5,                     !- Maximum Value of x",
+			"    ,                        !- Minimum Curve Output",
+			"    ,                        !- Maximum Curve Output",
+			"    Dimensionless,           !- Input Unit Type for X",
+			"    Dimensionless;           !- Output Unit Type",
+		});
+
+		ASSERT_FALSE( process_idf( idf_objects ) );
+
+		GetHumidifierInput();
+		ASSERT_EQ( 1, NumHumidifiers );
+		EXPECT_EQ( 1, Humidifier( 1 ).EfficiencyCurvePtr );
+
+	}
+
+	TEST_F( HumidifiersFixture, ThermalEfficiency ) {
+		// tests thermal efficiency modifier curve use
+
+		HumidifierData thisHum;
+
+		TimeStepSys = 0.25;
+		SysSizingRunDone = true;
+		CurSysNum = 1;
+
+		NumElecSteamHums = 0;
+		NumGasSteamHums = 1;
+		NumHumidifiers = 1;
+		Humidifier.allocate( NumGasSteamHums );
+		thisHum.HumType_Code = 2;
+		thisHum.NomCapVol = 4.00E-5;
+		thisHum.NomCap = 4.00E-2;
+		thisHum.NomPower = 103720.0;
+		thisHum.ThermalEffRated = 0.80;
+		thisHum.FanPower = 0.0;
+		thisHum.StandbyPower = 0.0;
+		thisHum.SchedPtr = ScheduleAlwaysOn;
+		thisHum.SchedPtr = ScheduleAlwaysOn;
+
+		FinalSysSizing.allocate( CurSysNum );
+		FinalSysSizing( CurSysNum ).MixTempAtCoolPeak = 20.0;
+		FinalSysSizing( CurSysNum ).MixHumRatAtCoolPeak = 0.00089;
+		FinalSysSizing( CurSysNum ).DesMainVolFlow = 1.60894;
+		FinalSysSizing( CurSysNum ).HeatMixHumRat = 0.05;
+		FinalSysSizing( CurSysNum ).CoolSupHumRat = 0.07;
+		FinalSysSizing( CurSysNum ).HeatSupHumRat = 0.10;
+
+		// calculate gas use rate and energy at full load
+		thisHum.AirInMassFlowRate = 1.8919;
+		thisHum.AirInTemp = 20.0;
+		thisHum.AirInEnthalpy = 25000.0;
+		thisHum.InletWaterTempOption = 1;
+		thisHum.CurMakeupWaterTemp = 20.0;
+		OutBaroPress = 101325.0;
+
+		std::string const idf_objects = delimited_string({
+			"Version,8.3;",
+			"  Curve:Quadratic,",
+			"    ThermalEfficiencyFPLR,   !- Name",
+			"    0.9375,                  !- Coefficient1 Constant",
+			"    0.0625,                  !- Coefficient2 x",
+			"    -7.0E-15,                !- Coefficient3 x**2",
+			"    0.0,                     !- Minimum Value of x",
+			"    1.2;                     !- Maximum Value of x",
+		});
+
+		ASSERT_FALSE( process_idf( idf_objects ) );
+
+		thisHum.EfficiencyCurvePtr = CurveManager::GetCurveIndex( "THERMALEFFICIENCYFPLR" );
+		thisHum.EfficiencyCurveType = CurveManager::Quadratic;
+
+		thisHum.CalcGasSteamHumidifier( 0.030 );
```

Comment:
@Nigusse I changed this test to use the new input processing capabilities. The assertions are the same as you had  and the only thing I changed was remove the IDD and IDF data structures creation. Instead it now uses process_idf() and the IDF snippet. Make sure this test is still what you intended, but I think it is.